File: COUNT-HILLS-VALLEYS/chv.go
package main

import (
	"fmt"
)

func countHillValley(nums []int) int {
	if len(nums) == 0 {
		return 0
	}

	arr := []int{nums[0]}
	for _, val := range nums {
		if arr[len(arr)-1] != val {
			arr = append(arr, val)
		}
	}

	count := 0
	for i := 1; i < len(arr)-1; i++ {
		if (arr[i] > arr[i-1] && arr[i] > arr[i+1]) || (arr[i] < arr[i-1] && arr[i] < arr[i+1]) {
			count++
		}
	}

	return count
}

func main() {
	fmt.Println(countHillValley([]int{2, 4, 1, 1, 6, 5}))
}

File: COUNT-HILLS-VALLEYS/chv.py
class Solution:
    def countHillValley(self, nums: List[int]) -> int:
        arr = [nums[0]]
        count = 0
        for i in nums[1:]:
            if i != arr[-1]:
                arr.append(i)
        for i in range(1, len(arr) - 1):
            if arr[i] < arr[i-1] and arr[i] < arr[i + 1]:
                count += 1
            elif arr[i] > arr[i-1] and arr[i] > arr[i + 1]:
                count += 1

        return count       